theories/All.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+set -u
+set -o pipefail
```

Comment:
In Rocq CI, we now get CI failures like this one:

```
File "theories/dune", line 11, characters 0-114:
11 | (rule
12 |  (targets All.v)
13 |  (deps All.sh (source_tree .))
14 |  (action (with-stdout-to %{targets} (run env sh ./All.sh))))
(cd _build/default/theories && /usr/bin/env sh ./All.sh) > _build/default/theories/All.v
./All.sh: 4: set: Illegal option -o pipefail
Done: 27% (830/3045, 2215 left) (jobs: 0)
                                         
Done: 34% (1112/3192, 2080 left, 1 failed) (jobs: 0)
                                                    
Done: 34% (1162/3376, 2214 left, 1 failed) (jobs: 0)
```

It seems related to this line.